src/core/server/ui_settings/types.ts

```diff
@@ -123,8 +131,9 @@ export interface UiSettingsParams {
 }
 
 export interface StringValidation {
-  regexString: string;
+  regex?: RegExp;
```

Comment:
Couldn't we support only one of those here?
If not, I think the right type would be 

```

export type StringValidation = {
  message: string;
  regex?: RegExp;
} | {
  message: string;
  regexString?: string;
}
```